models/reports/sql/generic/hospital-admissions-by-area-summary.sql

```diff
@@ -0,0 +1,87 @@
+        count(*) filter (where alh.discharge and alh.start_datetime::date between rm.month and (rm.month + '1 month'::interval - '1 day'::interval)) as discharges,
+        count(*) filter (where alh.death and alh.start_datetime::date between rm.month and (rm.month + '1 month'::interval - '1 day'::interval)) as deaths,
+        count(*) filter (where alh.transfer_in and alh.start_datetime::date between rm.month and (rm.month + '1 month'::interval - '1 day'::interval)) as transfer_ins,
+        count(*) filter (where alh.transfer_out and alh.start_datetime::date between rm.month and (rm.month + '1 month'::interval - '1 day'::interval)) as transfer_outs,
+        round(avg(alh.length_of_stay) filter (where alh.end_datetime::date between rm.month and (rm.month + '1 month'::interval - '1 day'::interval)), 1) as avg_length_of_stay
+    from reporting_months rm
+    join {{ ref('int__admission_history_location') }} alh
+        on alh.start_datetime::date <= (rm.month + '1 month'::interval - '1 day'::interval)
+        and (alh.end_datetime::date is null or alh.end_datetime::date >= rm.month)
+    join area_capacity lg on lg.location_group_id = alh.location_group_id
+    where rm.month <= current_date
+    group by
+        rm.month,
+        alh.facility_id,
+        alh.facility,
+        alh.location_group_id,
+        alh.location_group,
+        lg.capacity
+)
+
+select
+    to_char(lg.month, '{{ var("monthyear_format") }}') as "{{ translate_string('reportMonth', 'Month') }}",
+    lg.facility as "{{ translate_string('facilityName', 'Facility') }}",
+    lg.location_group as "{{ translate_string('areaName', 'Area') }}",
+    coalesce(lg.admissions, 0) as "{{ translate_string('admissionCount', 'Number of admissions') }}",
+    coalesce(lg.discharges, 0) as "{{ translate_string('dischargeCount', 'Number of discharges') }}",
+    coalesce(lg.deaths, 0) as "{{ translate_string('deathCount', 'Number of deaths') }}",
+    coalesce(lg.transfer_ins, 0) as "{{ translate_string('transfersIntoLocationCount', 'Number of transfers into location') }}",
+    coalesce(lg.transfer_outs, 0) as "{{ translate_string('transfersOutOfLocationCount', 'Number of transfers out of location') }}",
+    coalesce(lg.avg_length_of_stay, 0) as "{{ translate_string('averageLengthOfStay', 'Average length of stay') }}",
+    coalesce(lg.occupancy, 0) as "{{ translate_string('patientDayCount', 'Number of patient days') }}",
+    case
+        when lg.occupancy notnull and lg.capacity notnull then
+            concat(
+                round(
+                    lg.occupancy / (
+                        lg.capacity * case
+                            when lg.month > (current_date - '1 month'::interval)
+                                then current_date - lg.month
+                            else (lg.month + '1 month'::interval)::date - lg.month
```

Comment:
This section fails on the first of each month because we'll be dividing by zero. By adding the one, it includes current date in the count which is the expected behaviour for bed occupancy.
```suggestion
                            when lg.month > (current_date - '1 month'::interval)
                                then current_date - lg.month
                            else (lg.month + '1 month'::interval)::date - lg.month
```